obspy/taup/tau_model.py

```diff
@@ -413,3 +371,206 @@ def getBranchDepths(self):
+            noDisconDepths <type 'list'> (of float!?)
+            radiusOfEarth <type 'float'>
+            ray_params <type 'numpy.ndarray'> (1D, float)
+            sMod <class 'obspy.taup.slowness_model.SlownessModel'>
+            sourceBranch <type 'int'>
+            source_depth <type 'float'>
+            spherical <type 'bool'>
+            tauBranches <type 'numpy.ndarray'> (2D, type TauBranch)
+
+            TauBranch
+            =========
+            DEBUG <type 'bool'>
+            botDepth <type 'float'>
+            dist <type 'numpy.ndarray'>
+            isPWave <type 'bool'>
+            maxRayParam <type 'float'>
+            minRayParam <type 'float'>
+            minTurnRayParam <type 'float'>
+            tau <type 'numpy.ndarray'>
+            time <type 'numpy.ndarray'>
+            topDepth <type 'float'>
+
+            SlownessModel
+            =============
+            DEBUG <type 'bool'>
+            DEFAULT_SLOWNESS_TOLERANCE <type 'float'>
+            PLayers <type 'numpy.ndarray'>
+            PWAVE <type 'bool'>
+            SLayers <type 'numpy.ndarray'>
+            SWAVE <type 'bool'>
+            allowInnerCoreS <type 'bool'>
+            criticalDepths <type 'numpy.ndarray'>
+            fluidLayerDepths <type 'list'> (of DepthRange)
+            highSlownessLayerDepthsP <type 'list'> (of DepthRange)
+            highSlownessLayerDepthsS <type 'list'> (of DepthRange)
+            maxDeltaP <type 'float'>
+            maxDepthInterval <type 'float'>
+            maxInterpError <type 'float'>
+            maxRangeInterval <type 'float'>
+            minDeltaP <type 'float'>
+            radiusOfEarth <type 'float'>
+            slowness_tolerance <type 'float'>
+            vMod <class 'obspy.taup.velocity_model.VelocityModel'>
+
+            VelocityModel
+            =============
+            cmbDepth <type 'float'>
+            default_cmb <type 'float'>
+            default_iocb <type 'float'>
+            default_moho <type 'int'>
+            iocbDepth <type 'float'>
+            isSpherical <type 'bool'>
+            layers <type 'numpy.ndarray'>
+            maxRadius <type 'float'>
+            minRadius <type 'int'>
+            modelName <type 'unicode'>
+            mohoDepth <type 'float'>
+            radiusOfEarth <type 'float'>
+        """
+        # a) handle simple contents
+        keys = ['cmbBranch', 'cmbDepth', 'debug', 'iocbBranch', 'iocbDepth',
+                'mohoBranch', 'mohoDepth', 'noDisconDepths', 'radiusOfEarth',
+                'ray_params', 'sourceBranch', 'source_depth', 'spherical']
+        arrays = dict([(key, getattr(self, key)) for key in keys])
+        # b) handle .tauBranches
+        i, j = self.tauBranches.shape
+        # the following only works if all branches have the same length in
+        # their array-type attributes, e.g. branch.tau has same length for all
+        # branches. this seems to be the case at least for the default models
```

Comment:
I mean, I could also store each branch separately in serialization, it's just a bit less elegant. But I guess I should do it nevertheless.. safer beats nice looks I guess?